### ISSUE TITLE
[Bug] save_step is literally ignored train_gpt_xtts.py for xtts_v2 - memory usage also increases as the number of steps increases within each epoch

### ISSUE THREAD

**Issue by FurkanGozukara** [coqui-ai/TTS#3213]
the first problem is the VRAM usage increases as the number of steps increases. is this expected? within each epoch. so when it reaches closes to the epoch completion it uses shared VRAM. then it saves a checkpoint and vram usage gets reduced and repeats the process 

this problem doesn't occur sometimes it looks like random memory leak

![image](https://github.com/coqui-ai/TTS/assets/19240467/620bb774-f2f7-4a1c-8c1e-53e73d9c743f)

### Describe the bug

Here my entire config. It still keeps generating a checkpoint after each epoch completed. This is slowing down hugely

1 epoch is 145 steps since batch size is 4 and i have 585 training files

Also this message have any importance?

`The OrderedVocab you are attempting to save contains a hole for index 5024, your vocabulary could be corrupted !`

I have gradient accumulation steps as 63 but it has 0 effect on number of steps for per epoch. how is this possible? 

![image](https://github.com/coqui-ai/TTS/assets/19240467/20ad69ee-79b8-45d8-a6bb-51ce9c1f1ce9)

![image](https://github.com/coqui-ai/TTS/assets/19240467/727c098c-e5a1-4c48-9ce4-d852e27cd451)

here config.yaml

[config.yaml.txt](https://github.com/coqui-ai/TTS/files/13343177/config.yaml.txt)

### Environment

```

F:\coqui_test\TTS>git log -1
commit ac3df409a646dd607b7f5e812e661697ee6b9a37 (HEAD -> dev, origin/dev, origin/HEAD)
Merge: f32a4657 b85536b2
Author: Eren Gölge <erogol@hotmail.com>
Date:   Mon Nov 13 14:32:56 2023 +0100

Microsoft Windows [Version 10.0.19045.3636]
(c) Microsoft Corporation. All rights reserved.

F:\coqui_test\TTS\venv\Scripts>activate

(venv) F:\coqui_test\TTS\venv\Scripts>pip freeze
absl-py==2.0.0
accelerate==0.24.1
aiohttp==3.8.6
aiosignal==1.3.1
alembic==1.12.1
antlr4-python3-runtime==4.9.3
anyascii==0.3.2
appdirs==1.4.4
asteroid-filterbanks==0.4.0
async-timeout==4.0.3
attrs==23.1.0
audioread==3.0.1
Babel==2.13.1
bangla==0.0.2
blinker==1.7.0
bnnumerizer==0.0.2
bnunicodenormalizer==0.1.6
cachetools==5.3.2
certifi==2023.7.22
cffi==1.16.0
charset-normalizer==3.3.2
clean-fid==0.1.35
click==8.1.7
clip-anytorch==2.5.2
colorama==0.4.6
coloredlogs==15.0.1
colorlog==6.7.0
contourpy==1.2.0
coqpit==0.0.17
cycler==0.12.1
Cython==0.29.30
dateparser==1.1.8
decorator==5.1.1
docker-pycreds==0.4.0
docopt==0.6.2
einops==0.6.1
encodec==0.1.1
filelock==3.13.1
Flask==2.3.3
flatbuffers==23.5.26
fonttools==4.44.0
frozenlist==1.4.0
fsspec==2023.6.0
ftfy==6.1.1
g2pkk==0.1.2
gitdb==4.0.11
GitPython==3.1.40
google-auth==2.23.4
google-auth-oauthlib==1.1.0
greenlet==3.0.1
grpcio==1.59.2
gruut==2.2.3
gruut-ipa==0.13.0
gruut-lang-de==2.0.0
gruut-lang-en==2.0.0
gruut-lang-es==2.0.0
gruut-lang-fr==2.0.2
hangul-romanize==0.1.0
huggingface-hub==0.19.0
humanfriendly==10.0
HyperPyYAML==1.2.2
idna==3.4
imageio==2.32.0
inflect==5.6.2
itsdangerous==2.1.2
jamo==0.4.1
jieba==0.42.1
Jinja2==3.1.2
joblib==1.3.2
jsonlines==1.2.0
jsonmerge==1.9.2
jsonschema==4.19.2
jsonschema-specifications==2023.7.1
julius==0.2.7
k-diffusion==0.0.16
kiwisolver==1.4.5
kornia==0.7.0
lazy_loader==0.3
librosa==0.10.0
lightning==2.1.1
lightning-utilities==0.9.0
llvmlite==0.40.1
Mako==1.3.0
Markdown==3.5.1
markdown-it-py==3.0.0
MarkupSafe==2.1.3
matplotlib==3.7.3
mdurl==0.1.2
mpmath==1.3.0
msgpack==1.0.7
multidict==6.0.4
networkx==2.8.8
nltk==3.8.1
num2words==0.5.13
numba==0.57.0
numpy==1.22.0
oauthlib==3.2.2
omegaconf==2.3.0
onnxruntime-gpu==1.16.2
optuna==3.4.0
packaging==23.1
pandas==1.5.3
Pillow==10.0.1
platformdirs==4.0.0
pooch==1.8.0
primePy==1.3
protobuf==4.23.4
psutil==5.9.6
pyannote.audio==3.0.1
pyannote.core==5.0.0
pyannote.database==5.0.1
pyannote.metrics==3.2.1
pyannote.pipeline==3.0.1
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycparser==2.21
pydub==0.25.1
Pygments==2.16.1
pynndescent==0.5.10
pyparsing==3.1.1
pypinyin==0.49.0
pyreadline3==3.4.1
pysbd==0.3.4
python-crfsuite==0.9.9
python-dateutil==2.8.2
pytorch-lightning==2.1.1
pytorch-metric-learning==2.3.0
pytz==2023.3.post1
PyYAML==6.0.1
referencing==0.30.2
regex==2023.10.3
requests==2.31.0
requests-oauthlib==1.3.1
resize-right==0.0.2
rich==13.6.0
rpds-py==0.12.0
rsa==4.9
ruamel.yaml==0.18.5
ruamel.yaml.clib==0.2.8
safetensors==0.4.0
scikit-image==0.22.0
scikit-learn==1.3.0
scipy==1.11.3
semver==3.0.2
sentencepiece==0.1.99
sentry-sdk==1.35.0
setproctitle==1.3.3
shellingham==1.5.4
six==1.16.0
smmap==5.0.1
sortedcontainers==2.4.0
soundfile==0.12.1
soxr==0.3.7
speechbrain==0.5.15
SQLAlchemy==2.0.23
sympy==1.12
tabulate==0.9.0
tensorboard==2.15.1
tensorboard-data-server==0.7.2
tensorboardX==2.6.2.2
threadpoolctl==3.2.0
tifffile==2023.9.26
tokenizers==0.13.3
torch==2.1.0+cu121
torch-audiomentations==0.11.0
torch-pitch-shift==1.2.4
torchaudio==2.1.0+cu121
torchdiffeq==0.2.3
torchmetrics==1.2.0
torchsde==0.2.6
torchvision==0.16.0+cu121
tqdm==4.64.1
trainer==0.0.31
trampoline==0.1.2
transformers==4.33.3
-e git+https://github.com/coqui-ai/TTS@ac3df409a646dd607b7f5e812e661697ee6b9a37#egg=TTS
typer==0.9.0
typing_extensions==4.8.0
tzdata==2023.3
tzlocal==5.2
umap-learn==0.5.4
Unidecode==1.3.7
urllib3==2.1.0
wandb==0.16.0
wcwidth==0.2.9
Werkzeug==3.0.1
yarl==1.9.2

(venv) F:\coqui_test\TTS\venv\Scripts>

```

**Comment by FurkanGozukara** [comment]
I am still testing but i  made a example video from generated test speech :)


https://github.com/coqui-ai/TTS/assets/19240467/022a27f2-f711-452c-9fd6-f5a948b66b34

**Comment by codepharmer** [comment]
Lol I thought it was normal for VRAM usage to increase throughout the epoch, but I guess not. Is it possible that there's some type of caching happening?

**Comment by FurkanGozukara** [comment]
> Lol I thought it was normal for VRAM usage to increase throughout the epoch, but I guess not. Is it possible that there's some type of caching happening?

i also noticed this

i never seen vram increased through an epoch in any other trainings i made

**Comment by codepharmer** [comment]
I noticed this while training on https://github.com/mozilla/TTS models (coqui-ai's predecessor), so it seems like this has been around for a while.

**Comment by erogol** [comment]
VRAM fluctuation is normal. save_step works fine on my part. If you can provide a colab or something for reproducing the problem, we can better debug.

**Comment by FurkanGozukara** [comment]
> VRAM fluctuation is normal. save_step works fine on my part. If you can provide a colab or something for reproducing the problem, we can better debug.

i am using everything default but on Windows 10
